src/encode.py

```diff
@@ -0,0 +1,45 @@
+import pandas as pd
+
+FEATURES_DEFAULT = ('lesson_id', 'exercise_code', 'learning_object', 'exercise_code_level')
+TIME_WINDOWS_DEFAULT = ('60s', '1h', '1d', '5d', '30d', '365d')
+COUNTERS_DEFAULT = ('attempts', 'wins')
+
+def encode_df(df, feature_cumcount=[]):
+    '''
+    Encodes the wins and fails cumulative counts in the given time windows and dummifies the categorical variables of the dataset.
+    '''
+    for feature in FEATURES_DEFAULT:
```

Comment:
you might want to pass FEATURES_DEFAULT as a parameter, i think it's cleaner and way easier if you want to test the function

Reply:
corrected